### PR TITLE
Homework projects

### PR DESCRIPTION
**Pull request by adamblanchard**
This PR is based on https://github.com/HackYourFuture-CPH/JavaScript/pull/185 which moves https://github.com/HackYourFuture-CPH/js-projects into the js module.

There's quite a few changes, but they are organised by commit, so hopefully that helps!

Outstanding todos:

- [x] Add guide on deployment (read below)
- [x] Add screenshots/wireframes for each project @gabigviana @Will-hyf 
- [x] Archive js-projects once this has been merged

**To anyone reading:**
We'd like deployment to be a requirement of the js projects so trainees leave with something "live" that they can show on their portfolio. Since they have not passed db, node etc. i think https://github.com/HackYourFuture-CPH/hyf-project-template might be overkill? If so, can anyone write a few notes on what a deployment guide could look like for these projects?

### CODE REVIEW

**Review by vdaguenet — APPROVED** [review]

File: homework-projects/guides/demo-day-presentation.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+### Introduction (2 minutes)
+Briefly introduce your chosen project. (Refer to the "projects" folder for details.)
+
+### Weekly Task Breakdown (10 minutes)
+
+- JS2 Week 1: Discuss how you integrated Browser environment, DOM manipulation, and DOM event listeners into your project.
+- JS2 Week 2: Describe your use of Array functions and Arrow functions.
+- JS2 Week 3: Explain your implementation of Callback functions, Asynchronicity, and Scope.
+- JS3 Week 1: Share how you worked with JSON, APIs, and Fetch.
+- JS3 Week 2: Talk about your application of Promises and Async/Await.
```
Comment (jason-vasilev) [inline homework-projects/guides/demo-day-presentation.md]: Is the order of weeks correct? Where does it come from?

File: homework-projects/guides/demo-day-presentation.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+For each week, highlight:
```
Comment (jason-vasilev) [inline homework-projects/guides/demo-day-presentation.md]: [suggestion] Not a native English speaker, but somehow it sounds better to me if it's inversed:
```suggestion
Highlight for each week:
```

File: homework-projects/guides/demo-day-presentation.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+### Feedback and Discussion Session (25 minutes)
+
+After your presentation, there will be an open session for feedback and discussion. This is a valuable time to gain insights from your mentors and peers.
+
+Tips for a Successful Presentation:
+- Be Clear and Concise: Focus on the key points and challenges of your project.
+- Reflect on Your Learning Journey: Share how your understanding evolved over the weeks.
+- Be Open to Feedback: This is a learning opportunity – embrace the insights and suggestions from mentors and peers.
+
+This is a suggested guideline for you to follow. We look forward to seeing your projects and hearing about your journey through the JS2 and JS3 modules in a way you feel most appropriate. Good luck!
```
Comment (jason-vasilev) [inline homework-projects/guides/demo-day-presentation.md]: [suggestion] Not native EN speaker, but maybe `convenient` is more accurate than `appropriate` here?
```suggestion
This is a suggested guideline for you to follow. We look forward to seeing your projects and hearing about your journey through the JS2 and JS3 modules in a way you feel most convenient. Good luck!
```

File: javascript2/week1/readme.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ Event listeners
 
 ## Relevant links
 * [Preparation](preparation.md)
-* [Homework](homework.md)
+* [Homework](/homework-projects/readme.md)
```
Comment (jason-vasilev) [inline javascript2/week1/readme.md:22]:
```suggestion
* [Homework](/homework-projects/README.md)
```

File: javascript2/week2/readme.md [hunk 5]
```diff
@@ -9,7 +9,7 @@
 -->
 ## Relevant links
 * [Preparation](preparation.md)
-* [Homework](homework.md)
+* [Homework](/homework-projects/readme.md)
```
Comment (jason-vasilev) [inline javascript2/week2/readme.md:12]:
```suggestion
* [Homework](/homework-projects/README.md)
```

File: javascript2/readme.md [hunk 6]
```diff
@@ -2,6 +2,6 @@
...
 | ---- | ----- | ----------- | -------- |------------ |
-| 1.   | Browser environment<br>DOM manipulation <br> DOM event listeners | [Preparation](week1/preparation.md) | [Homework](week1/homework.md) | [Lesson plan](week1/lesson-plan.md)|
-| 2.   | Array functions <br> [Arrow function](week2/readme.md#arrow-function) | [Preparation](week2/preparation.md) | [Homework](week2/homework.md) | [Lesson plan](week2/lesson-plan.md)|
-| 3.   | Callback function <br> Asyncronicity <br> Scope | [Preparation](week3/preparation.md) | [Homework](week3/homework.md) | [Lesson plan](week3/lesson-plan.md)|
+| 1.   | Browser environment<br>DOM manipulation <br> DOM event listeners | [Preparation](week1/preparation.md) | [Homework](/homework-projects/readme.md) | [Lesson plan](week1/lesson-plan.md)|
+| 2.   | Array functions <br> [Arrow function](week2/readme.md#arrow-function) | [Preparation](week2/preparation.md) | [Homework](/homework-projects/readme.md) | [Lesson plan](week2/lesson-plan.md)|
+| 3.   | Callback function <br> Asyncronicity <br> Scope | [Preparation](week3/preparation.md) | [Homework](/homework-projects/readme.md) | [Lesson plan](week3/lesson-plan.md)|
```
Comment (jason-vasilev) [inline javascript2/readme.md:7]: I see the README is renamed to uppercase, but many places where it's reffered to it's used as lowercase. Would it make sense to keep it consistent, even if the linking is not case-sensitive?
```suggestion
| 3.   | Callback function <br> Asyncronicity <br> Scope | [Preparation](week3/preparation.md) | [Homework](/homework-projects/README.md) | [Lesson plan](week3/lesson-plan.md)|
```

**Review by jason-vasilev — APPROVED** [review]
Great effort here and super nice optimization. Hopefully it's more clear and easy for the trainees to understand what's optional, and how to navigate the folders and exercises.

I've left a few minor suggestions about wording. No strong feelings, just things sounded a tiny bit better to me. Also have a concern with the upper to lower case with README.md if it wouldn't cause an issue, but couldn't test it without things getting merged.

Not relevant to the PR,, I noticed that the repo is 75MB when downloaded. The bulk is primary due to a 15MB GIF in JS3, week1. Maybe something to change/improve on in the future - WebM might be a good alternative, if someone still keeps the original asset.

File: javascript2/week1/optional-homework.md [hunk 7]
```diff
@@ -1,11 +1,7 @@
 > [!WARNING]
-> You do not need to complete these homeworks. We are currently testing new homeworks for JS2 and JS3. More info [here](https://github.com/HackYourFuture-CPH/js-projects).
+> These are optional homework exercises that you can complete on top of your [homework project](/homework-projects/readme.md), if you wish. These will not be reviewed.
```
Comment (magdazelena) [inline javascript2/week1/optional-homework.md]: when I'm a reviewer I usually don't mind looking through those additional homeworks. Maybe `These will not be reviewed` is a little strong? Maybe `These are not required to be reviewed, although feel free to ask your mentor to do so`? Or something along these lines... Some students are very word to word :)

File: homework-projects/guides/demo-day-presentation.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+- Any particular successes or insights you gained.
+
+### Project Summary (3 minutes)
+Summarize what you learned from this project and how it ties together the concepts learned over JS2 and JS3.
+
+### Feedback and Discussion Session (25 minutes)
+
+After your presentation, there will be an open session for feedback and discussion. This is a valuable time to gain insights from your mentors and peers.
+
+Tips for a Successful Presentation:
```
Comment (magdazelena) [inline homework-projects/guides/demo-day-presentation.md:24]: one more tip suggestion: be mindful of time :D Framing presentation in time is crucial both in HYF as well irl.

**Review by magdazelena — APPROVED** [review]
Looks good! I added some minor wording comments :)

Regarding deployment guide: 
1. I went through the linked guide in the PR description. I don't have any experience with `onrender` but from the brief look it seems like you can deploy a static site via the tool. It might seem a little complex, but at this point everything for the students is. If it works like this the guide could just skip the api and db part so they don't get overwhelmed. 
**Pros**:
- they will have to use it anyway, so they might as well have a first attempt
- there won't be bazillion deployment options in HYF 
- it's good not to try and shield them too much
**Cons**:
- potentially complicated and overwhelming
- if overwhelming, might require additional support from mentors

2. Github pages?
They create their own repo anyway that they merge feature branches to. From that you can create a GH page in a couple of clicks.
**Pros**: 
- This is likely a tool that one will use after the course has ended
- it is fairly simple, especially for static sites
**Cons**:
- another deployment platform
- ...?

File: javascript2/week1/optional-homework.md (hunk 7 above)
Comment (adamblanchard) [reply javascript2/week1/optional-homework.md]: That's nice to hear, yes I'll change the wording along the lines you suggest :)

File: homework-projects/guides/demo-day-presentation.md (hunk 1 above)
Comment (adamblanchard) [reply homework-projects/guides/demo-day-presentation.md]: This text comes from the notes that Aarhus used last time to plan the demo day. These weekly agendas seem to align with the curriculum readme's e.g. https://github.com/HackYourFuture-CPH/JavaScript/tree/main/javascript2

Saying that, I think it's redundant to duplicate the topics here. I'll remove them and refer to the module readme(s)

File: javascript2/readme.md (hunk 6 above)
Comment (adamblanchard) [reply javascript2/readme.md:7]: Oo good catch! It looks like we use lower case for readme's in the rest of this repo. I'm thinking to rename the file for consistency instead.

File: homework-projects/guides/demo-day-presentation.md (hunk 3 above)
Comment (adamblanchard) [reply homework-projects/guides/demo-day-presentation.md]: How about "in a way that works for you and your project"?

File: homework-projects/guides/demo-day-presentation.md (hunk 2 above)
Comment (adamblanchard) [reply homework-projects/guides/demo-day-presentation.md]: I think the original sounds better. English is my first language, but it should sound good to everyone! I'll think if there's a different way to phrase it :)

**Comment by adamblanchard** [comment]
@magdazelena Thanks for your perspective on the deployment part! In that case, I added a small comment in the readme about deployment being a requirement and by default suggesting hyf-project-template. However, it can be agreed upon differently by staff/mentors later on.

It would be ideal to try the template, and tweak it to possibly be more simple, rather than introduce another method in my opinion. But it might be a bit of a learning hurdle for mentors the first time we really try it out.

**Pull request merged**